base/compiler/effects.jl

```diff
@@ -171,6 +171,60 @@ function Effects(effects::Effects = _EFFECTS_UNKNOWN;
         nonoverlayed)
 end
 
+function better_effects(new::Effects, old::Effects)
+    any_improved = false
+    if new.consistent == ALWAYS_TRUE
+        any_improved |= old.consistent != ALWAYS_TRUE
+    elseif new.consistent != old.consistent
```

Comment:
```suggestion
    elseif !iszero(new.consistent & CONSISTENT_IF_NOTRETURNED)
        old.consistent == ALWAYS_TRUE && return false
        any_improved |= iszero(old.consistent & CONSISTENT_IF_NOTRETURNED)
    elseif !iszero(new.consistent & CONSISTENT_IF_INACCESSIBLEMEMONLY)
        old.consistent == ALWAYS_TRUE && return false
        any_improved |= iszero(old.consistent & CONSISTENT_IF_INACCESSIBLEMEMONLY)
    elseif new.consistent != old.consistent
```
?

Reply:
Yeah, I suppose

Reply:
I guess I was thinking we might not need that one any more with our refinement logic, but while we have it, it should be here.